stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,186 @@
+
+    /**
+     * Folder to unzip from the compressed bytes.
+     */
+    private static final int FOLDER_TO_UNZIP_INDEX = 1;
+
+    /**
+     * Decompress/unzip byte arrays/blob.
+     *  @param fileContentAsByteArray file content as a byte arry
+     * @param outputFolder           destination folder
+     * @param folderToUnzip folder to unzip
+     */
+    protected static void decompress(byte[] fileContentAsByteArray, String outputFolder, String folderToUnzip) {
+        ZipInputStream zin = null;
+        try {
+            Path outdir = Paths.get(outputFolder);
+            zin = new ZipInputStream(new ByteArrayInputStream(fileContentAsByteArray));
+            ZipEntry entry;
+            String name, dir;
+            while ((entry = zin.getNextEntry()) != null) {
+                name = entry.getName();
+                if (!folderToUnzip.isEmpty() && name.startsWith(folderToUnzip)) {
+                    int index = name.lastIndexOf('/') + 1;
+                    name = name.substring(index);
+                }
+                    if (entry.isDirectory()) {
+                        mkdirs(outdir, name);
+                        continue;
+                    }
+                    // this part is necessary because file entry can come before directory entry where the
+                    // file is located
+                    dir = getDirectoryPath(name);
+                    if (dir != null) {
+                        mkdirs(outdir, dir);
+                    }
+                    extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            log.debug("I/O Exception when processing files ", e);
+            log.error("I/O Exception when processing files " + e.getMessage());
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                log.debug("I/O Exception when closing the input stream ", e);
+                log.error("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name   name of the file
+     */
+    private static void extractFile(ZipInputStream in, Path outdir, String name) {
+        byte[] buffer = new byte[4096];
+        BufferedOutputStream out = null;
+        try {
+            Path resourcePath = Paths.get(outdir.toString()).resolve(name);
+            out = new BufferedOutputStream(new FileOutputStream(resourcePath.toString()));
+            int count;
+            while ((count = in.read(buffer)) != -1) {
+                out.write(buffer, 0, count);
+            }
+        } catch (FileNotFoundException e) {
+            log.debug("File not found to process " + outdir, e);
+            log.error("File not found to process " + e.getMessage());
```

Comment:
I noticed all the exceptions in this class logged twice. Is there a specific reason?

Reply:
Removed the debug logs with  ad5c1ff125c11049a9b1c7e8ecb185e8e1e5ed02